src/Control/Monad/Freer/NonDet.hs

```diff
@@ -15,41 +15,33 @@ module Control.Monad.Freer.NonDet
   ) where
 
 import Control.Applicative (Alternative, (<|>), empty)
-import Control.Monad (msum)
+-- import Control.Monad (msum)
 
 import Control.Monad.Freer.Internal
-  ( Eff(..)
-  , Member
-  , NonDet(..)
-  , handleRelay
-  , prj
-  , qApp
-  , qComp
-  , tsingleton
-  )
+import Control.Monad.Freer.Interpretation
 
 -- | A handler for nondeterminstic effects.
 makeChoiceA
   :: Alternative f
   => Eff (NonDet ': effs) a
   -> Eff effs (f a)
-makeChoiceA = handleRelay (pure . pure) $ \m k ->
+makeChoiceA = relay (pure . pure) $ \m k ->
   case m of
     MZero -> pure empty
     MPlus -> (<|>) <$> k True <*> k False
 
 msplit
-  :: Member NonDet effs
+  :: () -- Member NonDet effs
   => Eff effs a
   -> Eff effs (Maybe (a, Eff effs a))
-msplit = loop []
-  where
-    loop jq (Val x) = pure (Just (x, msum jq))
-    loop jq (E u q) = case prj u of
-        Just MZero -> case jq of
-          []      -> pure Nothing
-          (j:jq') -> loop jq' j
-        Just MPlus -> loop (qApp q False : jq) (qApp q True)
-        Nothing    -> E u (tsingleton k)
-      where
-        k = qComp q (loop jq)
+msplit = undefined -- loop []
```

Comment:
Does anyone care about this? If so I'll figure out how to write it, but in the meantime cbf.

Reply:
No idea, I’ve never personally used the `NonDet` effect in any of my projects. To be honest, I don’t even know what this function is supposed to do.